lib/mayaUsd/render/vp2RenderDelegate/mesh.cpp

```diff
@@ -406,43 +380,54 @@ void HdVP2Mesh::Sync(
     if (HdChangeTracker::IsTopologyDirty(*dirtyBits, id)) {
         _meshSharedData._topology = GetMeshTopology(delegate);
 
-        // If there is uniform or face-varying primvar, we have to expand shared
-        // vertices in CPU because OpenGL SSBO technique is not widely supported
-        // on GPUs and 3D APIs.
-        bool requiresUnsharedVertices = false;
-        for (const auto& it : _meshSharedData._primvarSourceMap) {
-            const HdInterpolation interp = it.second.interpolation;
-            if (interp == HdInterpolationUniform ||
-                interp == HdInterpolationFaceVarying) {
-                requiresUnsharedVertices = true;
-                break;
-            }
-        }
+        const HdMeshTopology& topology = _meshSharedData._topology;
+        const VtIntArray& faceVertexIndices = topology.GetFaceVertexIndices();
+        const size_t numFaceVertexIndices = faceVertexIndices.size();
+
+        VtIntArray newFaceVertexIndices;
+        newFaceVertexIndices.resize(numFaceVertexIndices);
 
-        if (requiresUnsharedVertices) {
-            const HdMeshTopology& topology = _meshSharedData._topology;
+        if (_IsUnsharedVertexLayoutRequired(_meshSharedData._primvarSourceMap)) {
+            _meshSharedData._numVertices = numFaceVertexIndices;
+            _meshSharedData._renderingToSceneFaceVtxIds = faceVertexIndices;
 
             // Fill with sequentially increasing values, starting from 0. The
-            // new face vertex indices will then be implicitly used to assemble
-            // all primvar vertex buffers.
-            VtIntArray newFaceVtxIds;
-            newFaceVtxIds.resize(topology.GetFaceVertexIndices().size());
-            std::iota(newFaceVtxIds.begin(), newFaceVtxIds.end(), 0);
-
-            _meshSharedData._unsharedTopology.reset(
-                new HdMeshTopology(
-                    topology.GetScheme(),
-                    topology.GetOrientation(),
-                    topology.GetFaceVertexCounts(),
-                    newFaceVtxIds,
-                    topology.GetHoleIndices(),
-                    topology.GetRefineLevel()
-                )
-            );
+            // new face vertex indices will then be used to assemble all primvar
+            // vertex buffers.
+            std::iota(newFaceVertexIndices.begin(), newFaceVertexIndices.end(), 0);
```

Comment:
Is there anywhere we actually use the iota buffer? I saw in the algorithm that you don't actually use the buffer when you detect there is no re-ordering. Can we skip filling the buffer at all? You should add a comment here if we don't use the buffer when it goes through this code path, so that if someone changes things they know that they need to update the places using the buffer as well.

Reply:
The iota buffer is used to populate index buffer at https://github.com/Autodesk/maya-usd/blob/HdC-adsk/MAYA-105263/improve-dense-mesh-drawing-performance/lib/mayaUsd/render/vp2RenderDelegate/mesh.cpp#L862. I will add comment here to make it clear.